lib/components/molecules/Pagination/Pagination.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+/**
+ *
+ * Pagination
+ *
+ */
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+import styles from './Pagination.style';
+import type { Node } from './types';
+import { noop } from '@babel/types';
+
+type Props = {
+  size: number,
+  render: Function,
+  items: Array,
+  className: Array | string,
+};
+const buttons = (n, component) => Array.from('x'.repeat(n), (_, s) => component(s + 1));
+const Pagination = ({ size, render, items }: Props): Node => {
+  const [current, setCurrentPage] = useState(1);
+  const pages = Math.ceil(items.length / size);
+  const button = n => (
+    <button key={n} onClick={() => setCurrentPage(n)}>
+      Page {n}
+    </button>
+  );
+  const pageButtons = pages > 1 ? buttons(pages, button) : '';
+  const start = (current - 1) * size;
+  const end = start + size;
+  items = items.slice(start, end);
```

Comment:
You are not treating list in immutable way. Where you will get the next set of data if you overwrite the original source in the first pagination action itself.